lib/index.ts

```diff
@@ -105,11 +116,92 @@ export class Front {
 			reqType: 'GET' }, params, callback),
 	};
 
+	// Keys for Front access and event verification.
 	private apiKey: string;
+	private apiSecret: string;
 
-	constructor(apiKey: string) {
+	constructor(apiKey: string, apiSecret?: string) {
 		// Key.
 		this.apiKey = apiKey;
+		// Event signature secret.
+		if (apiSecret) {
+			this.apiSecret = apiSecret;
+		}
+	}
+
+	public registerEvents(opts: EventOptions, callback: EventCallback): Server | void {
+		let httpServer: Server| void;
+		let listener: express.Application;
+		const eventQueue: string[] = [];
+		const requestEvent = () => {
+			// Get next event in the queue.
+			const eventId = eventQueue[0];
+			this.httpCall({ path: 'events/<event_id>', reqType: 'GET' }, {
+				event_id: eventId
+			}).then((event: Event) => {
+				callback(null, event);
+			}).catch((err: FrontError) => {
+				callback(err);
+			}).finally(() => {
+				// Get another event if there is one, else finish.
+				eventQueue.shift();
+				if (eventQueue.length > 0) {
+					requestEvent();
+				}
+			});
+		};
+		const addToEventQueue = (id: string): void => {
+			// Push the event onto the queue. If there's nothing already
+			// on it, retrieve now, else just return.
+			eventQueue.push(id);
+			if (eventQueue.length === 1) {
```

Comment:
What's the use of the `if` here -- you just added an event above so it's surely `eventQueue > 0` -- is there a reason it mustn't be `>1` to call `requestEvent`?